ee/vellum_ee/workflows/display/nodes/vellum/inline_prompt_node.py

```diff
@@ -18,6 +18,23 @@
 _InlinePromptNodeType = TypeVar("_InlinePromptNodeType", bound=InlinePromptNode)
 
 
+def _has_nested_descriptors(blocks: Sequence[Union[PromptBlock, RichTextChildBlock]]) -> bool:
+    """Recursively check if any blocks contain BaseDescriptor instances."""
+    for block in blocks:
+        if isinstance(block, BaseDescriptor):
+            return True
+
+        if hasattr(block, "blocks") and hasattr(block, "block_type") and block.block_type == "CHAT_MESSAGE":
+            if _has_nested_descriptors(block.blocks):
+                return True
```

Comment:
```suggestion
        if block.block_type == "CHAT_MESSAGE" and _has_nested_descriptors(block.blocks):
            return True
```

here's a better way to do this. Do this on the Rich Text test below too